components/Segmented/src/Segmented/Segmented.cs

```diff
@@ -154,4 +155,16 @@ private void OnSelectedIndexChanged(DependencyObject sender, DependencyProperty
             _internalSelectedIndex = SelectedIndex;
         }
     }
+
+    private void OnOrientationChanged()
+    {
+        for (int i = 0; i < Items.Count; i++)
+        {
+            var container = ContainerFromIndex(i) as SegmentedItem;
+            if (container is null)
+                continue;
+
+            container.UpdateOrientation(Orientation);
```

Comment:
```suggestion
            if (ContainerFromIndex(i) is SegmentedItem item)
            {
                item.UpdateOrientation(Orientation);
            }
```
Would this be simpler?

Reply:
Yeah. I have a tendency to over user guard checks...